### DISCUSSION TITLE
Built with Docusaurus 3.5.2 on 19 Sep 2024.

### DISCUSSION THREAD

**Discussion by ilg-ul** [facebook/docusaurus#10512] [Q&A]
I just completed the migration of 20+ documentation sites to Docusaurus, and I'm working on streamlining the maintenance procedures for the whole group of projects.

One of the issues I identified is to know what Docusaurus version was used to build each site, and when it was built.

A possible solution would be to adjust the message shown in the footer, and add the version and the date (as in the subject of this topic).

I saw the copyright definition in the configuration file (in my case it is `Copyright © ${new Date().getFullYear()} Liviu Ionescu. Built with Docusaurus.`), but I don't know how to get the current Docusaurus version. Also, in this context, is the `new Date()` the date the static pages were built, or, by some React magic, the date the page is accessed?

Any suggestions? Anyone else interested in such a feature?

**Comment by allenscha** [comment]
You can view the version of Docusaurus used to build the site in the meta tag of the resulting HTML. (search with "generator" on the browser dev tools)

```
<meta name="generator" content="Docusaurus vx.y.z">
```

But it appears that the feature to make the Docusaurus version or build date displayed on the site is not available yet.

**Reply by slorber** [reply]
Yes this meta tag is there for a while already and can tell you the Docusaurus version by inspecting the DOM.

`new Date()` in the footer comes from the Node.js config file and is executed at build time, so it will show the date of the build.

What you can do:
- read version with [`useDocusaurusContext().siteMetadata.docusaurusVersion`](https://docusaurus.io/docs/docusaurus-core#useDocusaurusContext)
- `customFields.buildTime: new Date().getTime()`

I think we could add `siteMetadata.buildTime` or something to make it more convenient.

**Reply by ilg-ul** [reply]
Showing the build time was not difficult, adding it to the `footer.copyright` worked:

```
copyright: `Copyright © ${new Date().getFullYear()} Liviu Ionescu. Built with Docusaurus on ${new Date().toDateString()}.`,
```

However, attempts to add a similar expression for `useDocusaurusContext().siteMetadata.docusaurusVersion` failed, since adding `import useDocusaurusContext from '@docusaurus/useDocusaurusContext';` to `docusaurus.config.ts` failed (the https://docusaurus.io/docs/docusaurus-core#useDocusaurusContext page does not document an alternate way to use this hook).

In the end I hacked it by parsing the `package.json` file and extracting the version from the `@docusaurus/core` dependency, but I'm not very happy with this solution, perhaps you could provide a better one.

**Reply by slorber** [reply]
You can't use a React hook on the Node.js side (config). 

Either you read `package.json` on the Node side, or you create/swizzle a component and use the provided hook.

The config copyright line is a string created on Node.js side, not a React component.

**Reply by ilg-ul** [reply]
> You can't use a React hook on the Node.js side (config).

I expected this. So getting the version via `useDocusaurusContext().siteMetadata.docusaurusVersion` in the `docusaurus.config.ts` file is not functional.

**Comment by ilg-ul** [comment]
Wouldn't it be useful to display the Docusaurus version and build date by default in the footer?

**Reply by slorber** [reply]
You are the very first to ask for that. It's pretty easy to do it on your own.

```js
copyright: `Copyright © ${new Date().getFullYear()} - Version ${require('@docusaurus/core/package.json').version}`
```
